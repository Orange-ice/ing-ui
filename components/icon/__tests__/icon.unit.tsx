import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react';
import {Icon} from '../../index';

describe('Icon 组件测试', () => {
  it('快照测试', () => {
    const dom = renderer.create(<Icon name="tiger"/>).toJSON();
    expect(dom).toMatchSnapshot();
  });

  it('name props 测试', () => {
    const {container} = render(<Icon name="tiger"/>);
    expect(container.querySelector('svg use')).not.toEqual(null);
    expect(container.querySelector('svg use')!.getAttribute('xlink:href')).toEqual('#tiger');
  });

  it('click 事件测试', () => {
    const onClick = jest.fn();
    const {container} = render(<Icon name="tiger" onClick={onClick}/>);
    fireEvent.click(container.querySelector('svg') as SVGElement);
    expect(onClick).toBeCalledTimes(1);
  });
});
