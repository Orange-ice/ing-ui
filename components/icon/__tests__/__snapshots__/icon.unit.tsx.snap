// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 组件测试 快照测试 1`] = `
<svg
  className="ice-icon"
>
  <use
    xlinkHref="#tiger"
  />
</svg>
`;
